### ISSUE TITLE
Julia language server cannot start

### ISSUE THREAD

**Issue by roiholtzman** [emacs-lsp/lsp-mode#2722]
The Julia language server, `lsp-julia` cannot start.

I use doom emacs on MacOS. I upgraded doom and the problem continues. 
A surprising thing is that initially everything worked great. Then I restarted emacs and the problem appeared. Using some help on discord I upgraded doom emacs, but still things are not working.

The error from `Julia-ls::stderr`:
```
ERROR: MethodError: no method matching LanguageServer.FoldingRangeClientCapabilities(::Nothing)
Closest candidates are:
  LanguageServer.FoldingRangeClientCapabilities(::Any, !Matched::Any, !Matched::Any) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/protocol/features.jl:217
  LanguageServer.FoldingRangeClientCapabilities(; dynamicRegistration, rangeLimit, lineFoldingOnly) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/interface_def.jl:50
  LanguageServer.FoldingRangeClientCapabilities(!Matched::Union{Missing, Bool}, !Matched::Union{Missing, Int64}, !Matched::Union{Missing, Bool}) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/protocol/features.jl:217
  ...
Stacktrace:
 [1] LanguageServer.TextDocumentClientCapabilities(::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/interface_def.jl:56
 [2] LanguageServer.ClientCapabilities(::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/interface_def.jl:56
 [3] LanguageServer.InitializeParams(::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/protocol/initialize.jl:111
 [4] dispatch_msg(::JSONRPC.JSONRPCEndpoint{Base.PipeEndpoint,Base.PipeEndpoint}, ::JSONRPC.MsgDispatcher, ::Dict{String,Any}) at /Users/roiholtzman/.julia/packages/JSONRPC/xbAlZ/src/typed.jl:64
 [5] run(::LanguageServerInstance) at /Users/roiholtzman/.julia/packages/LanguageServer/y1ebo/src/languageserverinstance.jl:308
 [6] top-level scope at none:1

Process julia-ls stderr finished
```

in the `lsp-log` there is:
```
Command "julia --startup-file=no --history-file=no -e using InteractiveUtils, Sockets, SymbolServer, LanguageServer; server = LanguageServer.LanguageServerInstance( stdin, stdout,"/Users/roiholtzman/Projects/07_reset_systems/07_code/LocalResetExclusionProcessWatson/","", nothing, "/Users/roiholtzman/.julia/symbolstorev2-lsp-julia"); run(server);" is present on the path.
Command "julia --startup-file=no --history-file=no -e using InteractiveUtils, Sockets, SymbolServer, LanguageServer; server = LanguageServer.LanguageServerInstance( stdin, stdout,"/Users/roiholtzman/Projects/07_reset_systems/07_code/LocalResetExclusionProcessWatson/","", nothing, "/Users/roiholtzman/.julia/symbolstorev2-lsp-julia"); run(server);" is present on the path.
Command "julia --startup-file=no --history-file=no -e using InteractiveUtils, Sockets, SymbolServer, LanguageServer; server = LanguageServer.LanguageServerInstance( stdin, stdout,"/Users/roiholtzman/Projects/07_reset_systems/07_code/LocalResetExclusionProcessWatson/","", nothing, "/Users/roiholtzman/.julia/symbolstorev2-lsp-julia"); run(server);" is present on the path.
Found the following clients for /Users/roiholtzman/Projects/07_reset_systems/07_code/LocalResetExclusionProcessWatson/src/dynamics.jl: (server-id julia-ls, priority 0)
The following clients were selected based on priority: (server-id julia-ls, priority 0)
```

**Comment by jcs090218** [comment]
I think this issue is resolved base on the thread here, https://github.com/non-Jedi/lsp-julia/issues/46?


Close this now. :)

**Comment by devmotion** [comment]
I think this issue is still not fixed. It seems that setting `(setq lsp-enable-folding t)` is merely a workaround but does not fix the underlying issue. [According to the LanguageServer.jl devs](https://github.com/julia-vscode/LanguageServer.jl/issues/844#issuecomment-810563376) the problem is a violation of the specs by the client, probably due to a bug lsp-mode.

Since unfortunately the workaround contained in Doom Emacs does not work ([it seems one has to close the Julia file when the LS crashes and reopen it](https://github.com/gdkrmr/lsp-julia/issues/35#issuecomment-1102504324)), I tried to fix the problem. I am not familiar with the code base in lsp-mode and a beginner when it comes to lisp programming but just modifying the folding range capabilities such that they match the other conditional capabilities in lsp-model.el seemed to be sufficient: https://github.com/emacs-lsp/lsp-mode/compare/master...devmotion:lsp-mode:dw/fix_foldingrange I did not experience any crashes anymore. However, unfortunately, I'm not sure if that is the correct fix and how to add a test for it :grimacing:
